torch/testing/_internal/common_methods_invocations.py

```diff
@@ -10800,6 +10943,31 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
            ),
            supports_expanded_weight=True,
            supports_out=False,),
+    OpInfo('nn.functional.conv3d',
+           aliases=('conv3d',),
+           aten_name='conv3d',
+           dtypes=floating_and_complex_types_and(torch.int64, torch.bfloat16),
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16, torch.chalf,
+                                                       *[torch.bfloat16] if (CUDA11OrLater or TEST_WITH_ROCM) else []),
+           sample_inputs_func=sample_inputs_conv3d,
+           error_inputs_func=error_inputs_conv3d,
+           gradcheck_nondet_tol=GRADCHECK_NONDET_TOL,
```

Comment:
Is the backward of this function non-deterministic?

Reply:
Thanks for looking into it, @lezcano! Yes, its backward function (and the function itself) is non-deterministic for CUDA >= 10.2 and when using cuBlas on multiple streams.

A list of functions that uses deterministic algorithms is listed [here](https://pytorch.org/docs/stable/generated/torch.use_deterministic_algorithms.html).

I got a warning message when I tried to use the deterministic algorithm. Please see:
<details>
<summary>script</summary>

<p>

```python
In [1]: import torch

In [2]: torch.set_deterministic_debug_mode(True)

In [3]: input = torch.rand((2, 2, 3, 4, 4), device='cuda')

In [4]: weight = torch.rand((2, 2, 3, 2, 4), device='cuda')

In [5]: bias = torch.rand((2,), device='cuda')

In [6]: torch.nn.functional.conv3d(input, weight, bias, groups=1).backward()
<ipython-input-6-861dd68c93ac>:1: UserWarning: Deterministic behavior was enabled with either `torch.use_deterministic_algorithms(True)` or `at::Context::setDeterministicAlgorithms(true)`, but this operation is not deterministic because it uses CuBLAS and you have CUDA >= 10.2. To enable deterministic behavior in this case, you must set an environment variable before running your PyTorch application: CUBLAS_WORKSPACE_CONFIG=:4096:8 or CUBLAS_WORKSPACE_CONFIG=:16:8. For more information, go to https://docs.nvidia.com/cuda/cublas/index.html#cublasApi_reproducibility (Triggered internally at /home/khushi/Documents/pytorch/aten/src/ATen/Context.cpp:143.)
  torch.nn.functional.conv3d(input, weight, bias, groups=1).backward()
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Input In [6], in <cell line: 1>()
----> 1 torch.nn.functional.conv3d(input, weight, bias, groups=1).backward()

File ~/Documents/pytorch/torch/_tensor.py:482, in Tensor.backward(self, gradient, retain_graph, create_graph, inputs)
    472 if has_torch_function_unary(self):
    473     return handle_torch_function(
    474         Tensor.backward,
    475         (self,),
   (...)
    480         inputs=inputs,
    481     )
--> 482 torch.autograd.backward(
    483     self, gradient, retain_graph, create_graph, inputs=inputs
    484 )

File ~/Documents/pytorch/torch/autograd/__init__.py:197, in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables, inputs)
    192     retain_graph = create_graph
    194 # The reason we repeat same the comment below is that
    195 # some Python versions print out the first line of a multi-line function
    196 # calls in the traceback and some print out the last line
--> 197 Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
    198     tensors, grad_tensors_, retain_graph, create_graph, inputs,
    199     allow_unreachable=True, accumulate_grad=True)

RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```

</p>
</details>

Thanks!